packages/react/src/components/table/table.tsx

```diff
@@ -140,31 +142,55 @@ const ExpandButton = styled(IconButton) <{ $expanded: boolean }>`
     }
 `;
 
-function getUtilityColumn<T extends object>(type: UtilityColumnType, t: TFunction<'translation'>): TableColumn<T> {
+const RadioButton = styled.input`
+    margin: 3px 3px 3px 5px;
+    vertical-align: inherit;
+`;
+
+function getUtilityColumn<T extends object>(
+    type: UtilityColumnType,
+    t: TFunction<'translation'>,
+    selectionMode?: SelectionMode,
+): TableColumn<T> {
     const column: TableColumn<T> = {
         id: type,
         className: utilColumnClassName,
     };
 
     switch (type) {
         case 'selection':
-            column.header = ({ table }) => (
-                <Checkbox
-                    data-testid="row-checkbox-all"
-                    checked={table.getIsAllRowsSelected()}
-                    indeterminate={table.getIsSomeRowsSelected()}
-                    onChange={table.getToggleAllRowsSelectedHandler()}
-                />
-            );
-            column.cell = ({ row }) => (
-                <Checkbox
-                    data-testid={`row-checkbox-${row.index}`}
-                    checked={row.getIsSelected()}
-                    disabled={!row.getCanSelect()}
-                    indeterminate={row.getIsSomeSelected()}
-                    onChange={row.getToggleSelectedHandler()}
-                />
-            );
+            if (selectionMode === 'multiple') {
+                column.header = ({ table }) => (
+                    <Checkbox
+                        data-testid="row-checkbox-all"
+                        checked={table.getIsAllRowsSelected()}
+                        indeterminate={table.getIsSomeRowsSelected()}
+                        onChange={table.getToggleAllRowsSelectedHandler()}
+                    />
+                );
+                column.cell = ({ row }) => (
+                    <Checkbox
+                        data-testid={`row-checkbox-${row.index}`}
+                        checked={row.getIsSelected()}
+                        disabled={!row.getCanSelect()}
+                        indeterminate={row.getIsSomeSelected()}
+                        onChange={row.getToggleSelectedHandler()}
+                    />
+                );
+            } else if (selectionMode === 'single') {
+                column.cell = ({ table, row }) => (
+                    <RadioButton
```

Comment:
La radio button devrait avoir un `<label>` d'associé. S'il n'y a pas de `<label>` les technologies d'assistances (lecteur d'écran) vont seulement annoncer que c'est un radio button et rien d'autre. Le `<label>` sert à identifier de quel radio button il s'agit. _e.g._, si le `<label>` est '_patate_' le lecteur d'écran va annoncer: '_patate, radio button_'
Il y a deux options possibles:

1. **aria-label**
ajouter un `aria-label="label du radio button"` à l'input.
Tu aurais donc: 
`<input data-testid="..." type="radio" class="..." aria-label="patate">`.

2. **hidden text**
ajouter une classe qui vient cacher visuellement le `<label>`.
`<input id="idInput" data-testid="..." type="radio" class="..."><label for="idInput" class="visually-hidden">patate</label>`

Comme mentionné dans un autre commentaire, il faudrait utiliser le composant de radio button du DS. Ça va p-e demander du boulot dans le composant radio-button par contre.
Je vais regarder le travaille qu'il y a à faire sur le composant et si c'est trop de boulot, je vais créer une carte pour y faire les modifications et on reviendra sur ta PR quand les modifications pour le radio button seront fait.